File: Amina/Mahmood_PE02-Fixed/PE02_Java/MySQLDatabase.java
/***
   *Amina Mahmood
   *ISTE 330.01-02
   *Mr. Floser
   *Database Connectivity and Access
***/
import java.sql.*;

public class MySQLDatabase
{
      // Definitions for MySQL connection
      String uri = "jdbc:mysql://localhost/travel?autoReconnect=true&useSSL=false&useUnicode=true&useJDBCCompliantTimezoneShift=true&useLegacyDatetimeCode=false&serverTimezone=UTC";
      String driver = "com.mysql.cj.jdbc.Driver";
      String user = "root";
      String password ="student";
      
      Connection conn = null;
      
      public MySQLDatabase()
      {
//          // Connects and opens Database
//          if( connect() == true )
//          {
//             // Close Database and close connection
//             if(closeConnect() == true)
//             {
//                System.out.println("Closed the program!");
//             }
//          }
                  
       }
      
      // Connect method
      public boolean connect()
      {
         // Loading driver
         try
         {
            Class.forName(driver);
            //System.out.println("Driver loaded.");
            conn = DriverManager.getConnection(uri, user, password);
            //System.out.println("Database Opens");
            return true;
         }
         catch( ClassNotFoundException ce )
         {
            //System.out.println("Could not load a driver " + driver);
            ce.printStackTrace();
            System.exit(1);
            return false;
         }
         catch (SQLException e) 
         {
            //System.out.println("Could not connect to Database " + uri);
            e.printStackTrace();
            return false;
         }
      }
      
      // Close the database
      public boolean closeConnect()
      {
        try 
        {
            if (conn != null) 
            {
                conn.close();
                System.out.println("Database successfully closed.");
                return true;
            }
            else
            {
                return false;
            }
        } 
        catch (SQLException sqle) 
        {
            System.out.println("Could not close database");
            sqle.printStackTrace();
            return false;
        }
    }

}

File: Amina/Mahmood_PE02/PE02_CSharp/SQLServerDatabase/MySQLDatabase.cs
﻿/***
   *Amina Mahmood
   *ISTE 330.01-02
   *Mr. Floser
   *Database Connectivity and Access
***/
using MySql.Data.MySqlClient;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MySQLDatabase
{
    class MySQLDatabase
    {
        string mysqlConnStr = "server=127.0.0.1;Database=travel;uid=root;pwd=student;";
        MySqlConnection conn2;

        // Public method
        public MySQLDatabase()
        {
            try
            {
                if (connect() != false)
                {
                    // Open the database
                    Console.WriteLine("Database Opens");
                }

                // MySql close db
                close();
            }
            catch (MySqlException E)
            {
                Console.WriteLine("MySql Error: " + E.Message);
                Console.WriteLine("MySql " + E.StackTrace);
            }
        }

        // Connect method
        public Boolean connect()
        {
            try
            {
                conn2 = new MySqlConnection(mysqlConnStr);
                return true;
            }
            catch (MySqlException e)
            {
                Console.WriteLine("Could not connect to Database " + e);
                return false;
            }
        }

        // Close method
        public Boolean close()
        {
            try
            {
                if (conn2 != null)
                {
                    conn2.Close();
                    Console.WriteLine("Database successfully closed.");
                    return true;
                }
                else
                {
                    return false;
                }
            }
            catch (MySqlException e)
            {
                Console.WriteLine("Could not connect to Database " + e);
                return false;
            }
        }
    }
}
